### DISCUSSION TITLE
Transitive dependencies upgrade via script

### DISCUSSION THREAD

**Discussion by GiveUpBuddy** [renovatebot/renovate#32973] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab 17.5 + Renovate 39.57.1

### Please tell us more about your question or problem

Hi everyone! I’m pretty new to this and need help setting up Renovate. I’ve been trying to make it work, but I keep running into issues.

Here’s the situation: we’re using a self-hosted GitLab, and Renovate runs via a scheduled pipeline, handling multiple repositories listed in autodiscoverFilter.

Our developers want the following workflow: when Renovate starts, it should first run a script (the script is already ready and works fine) to generate transitive dependencies and save them in transitive.libs.versions.toml. After that, Renovate should analyze both files: libs.versions.toml and transitive.libs.versions.toml.

The issue is that Renovate only supports postUpgradeTasks, so I can’t figure out how to run the script first and then process both files in the same Renovate run. My workaround is to include the script in postUpgradeTasks, followed by another call to Renovate inside it (basically Renovate calling itself).

The problem? This second Renovate call needs a configuration file to know what to do. I can’t figure out how to make Renovate process the current repository and work on a branch like renovate/* instead of just the default branch.

It would make sense to run the script in a separate CI/CD job and then trigger Renovate, but since there are multiple repositories to handle, I have no idea how to set this up in GitLab CI.

Maybe I’m completely off track here and using the wrong approach or tools.

Any tips or guidance would be super helpful! Thanks!

### Logs (if relevant)

<details><summary>Logs</summary>

```

"postUpgradeTasks": {
    "commands": [
      "./gradlew listTransitiveDependencies",
      "renovate --token $GITHUB_COM_TOKEN"
    ],
    "fileFilters": [
      "transitive.libs.versions.toml",
      "renovate.log"
    ],
    "executionMode": "branch"
  },

```

</details>

**Comment by rarkins** [comment]
Unfortunately Renovate doesn't natively support this type of "pre-scan tasks"
